pyphare/pyphare/pharesee/geometry.py

```diff
@@ -245,28 +247,34 @@ def level_ghost_boxes(hierarchy):
 
     for ilvl, lvl in hierarchy.levels().items():
 
-        if not is_root_lvl(lvl):  # level ghost do not make sense for periodic root level
+        if is_root_lvl(lvl):  # level ghost do not make sense for periodic root level
+            continue
+
+        gaboxes_info = gaboxes[ilvl]
+
+        for info in gaboxes_info:
 
-            gaboxes_info = gaboxes[ilvl]
+            patch_data = info["pdata"]
+            ghostAreaBoxes = info["boxes"]
 
-            for info in gaboxes_info:
+            for gabox in ghostAreaBoxes:
 
-                patch_data = info["pdata"]
-                gaboxes = info["boxes"]
+                # now loop on all particle patchData
+                # keep only parts of the ghost boxes that do
+                # not intersect other patch data interior
 
-                for gabox in gaboxes:
+                if True:  # if periodic, always true for now
+                    refined_domain_box = hierarchy.refined_domain_box(ilvl)
+                    patches = get_periodic_list(lvl.patches, refined_domain_box)
 
-                    # now loop on all particle patchData
-                    # keep only parts of the ghost boxes that do
-                    # not intersect other patch data interior
+                for patch in patches:
 
-                    if True:  # if periodic, always true for now
-                        refined_domain_box = hierarchy.refined_domain_box(ilvl)
-                        patches = get_periodic_list(lvl.patches, refined_domain_box)
+                    particles_ids = [key for key in patch.patch_datas.keys() if key.endswith("particles")]
 
-                    for patch in patches:
+                    for particles_id in particles_ids:
```

Comment:
ok I think same as the above function, these level ghost boxes are going to be the same for all populations so I would say there is no need to loop over them all.

However I know see that the dict we return also gets the patch data which is going to be different for each pop (same applies to the function above actually...)

so... it seems stupid to get Npop times the same boxes, but it also seems stupid to get only 1 patchdata... (why that one ?)

Reply:
I don't think particles_ids is ever > 1 in size

Reply:
you mean in this test but in general yes ?

Reply:
I think in general, will confirm

Reply:
I don't see why it would be only size one, it should be equal to the total number of populations in the simulation.
We either need to, return all ghost boxes for all populations, or assume that the caller of the function knows that there's only one ghost box for all populations. I'm not sure which is simpler, because for the latter on the caller side you have to go through the patch_datas again and check if they are particles, but if you just give back a box per patch_data of type particles, you know its' the right one

Reply:
how it is here is, a ghost box per patch_data, so if that's the what's decided, I don't think anything needs to change

Reply:
Technically the function returns the level ghost boxes, which a priori has nothing to do with particle populations since per definition particle ghost width is invariant with respect to populations. The caller should know that I think.
This being said, we can also keep it the way it is now and see if that's inconvenient at some point.